### ISSUE TITLE
Getting Concatenated Field Value in Permalinks

### ISSUE THREAD

**Issue by puremana** [athlan/wordpress-custom-fields-permalink-plugin#43]
We are having an issue where some of the generated permalinks are having the custom locale field concatenated.

**Permalinks Settings Code**
`/%field_locale%/%postname%/`

**Custom Locale Field Value**
![image](https://user-images.githubusercontent.com/2869411/54732290-e0152c00-4bf7-11e9-8cb9-1384584db6ea.png)

**Expected**
![image](https://user-images.githubusercontent.com/2869411/54732217-76951d80-4bf7-11e9-8739-c51a9723c6cb.png)

**Reality**
![image](https://user-images.githubusercontent.com/2869411/54732234-9298bf00-4bf7-11e9-999a-1bb3f822ca08.png)

We are on version 2.0.

This is very inconsistent and rarely happens. The affected locales seem different each time too.

I hope this is enough information to diagnose the problem, happy to provide more if needed

**Comment by athlan** [comment]
Hello, thanks for contact. Could you please check in the database, whether your custom field `locale` for thi particular post does not have multiple `en-ca` values?

**Comment by puremana** [comment]
Yes, I found that if I create a duplicate locale value in the `wpdb_postmeta` table it concatenates them

![image](https://user-images.githubusercontent.com/2869411/54788126-e1dffd80-4c92-11e9-8758-bfe2be8baf42.png)

![image](https://user-images.githubusercontent.com/2869411/54788136-e60c1b00-4c92-11e9-8eec-febd4467174f.png)

**Comment by athlan** [comment]
That' the issue. Plugin concatenates duplicated values by default.

If you want to have the single value, you can:
* See what https://github.com/athlan/acf-permalink offers
* [or] Use hooks to remove duplicates: https://github.com/athlan/wordpress-custom-fields-permalink-plugin/wiki/Plugin-hooks
* [or] Specify the separator joining values using aforementioned hooks or acf extension

**Comment by puremana** [comment]
I can't see any way to format a text field to only show one value, would you be open to a pull request to fix this?

**Comment by athlan** [comment]
Hello,

It's already supported in plugin. [`wpcfp_get_post_metadata`](https://github.com/athlan/wordpress-custom-fields-permalink-plugin/wiki/Plugin-hooks#wpcfp_get_post_metadata) filter should do a trick for you. You'll receive all post params and then you can assign the single one to array key which will be used.

But anyway, it's little strange that you're facing such issue, because plugin [should take the very first value of meta field](https://github.com/athlan/wordpress-custom-fields-permalink-plugin/blob/1076a791348c3b6f22b5555874805e661c9b2feb/includes/class-wp-permalink.php#L115), we have even [test for that](https://github.com/athlan/wordpress-custom-fields-permalink-plugin/blob/master/test/integration/suites/MetaKeyPermalinkStructure/PostWithDuplicatedMetaKey.php).

1. Could you also pase here output of
```
var_dump(get_post_meta($post->ID)));
```
of your case?
2. Could you share permalink structure?

**Comment by puremana** [comment]
2. ![image](https://user-images.githubusercontent.com/2869411/54902226-b03e8f00-4f3d-11e9-93b0-7e6efa254423.png)

Which resulted in 
![image](https://user-images.githubusercontent.com/2869411/54902310-e419b480-4f3d-11e9-9860-d4dba1837f6e.png)

I can't get that output currently, but tomorrow I should be able to

**Comment by puremana** [comment]
Different recipe, same issue, here's the var_dump output

![image](https://user-images.githubusercontent.com/2869411/54958499-b292fe80-4fba-11e9-8e7f-29f7bfab5f93.png)

**Comment by puremana** [comment]
@athlan If the plugin is supposed to take the first value, doesn't this make it a bug?
